src/main/java/fi/metatavu/keycloak/graphapi/GraphApiUserManagerIdentityProviderMapper.java

```diff
@@ -0,0 +1,220 @@
+package fi.metatavu.keycloak.graphapi;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.protobuf.MapEntry;
+import fi.metatavu.keycloak.graphapi.client.GraphApiClient;
+import fi.metatavu.keycloak.graphapi.model.GraphUser;
+import org.jboss.logging.Logger;
+import org.keycloak.broker.provider.AbstractIdentityProviderMapper;
+import org.keycloak.broker.provider.BrokeredIdentityContext;
+import org.keycloak.models.IdentityProviderMapperModel;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.UserModel;
+import org.keycloak.provider.ProviderConfigProperty;
+import org.keycloak.representations.AccessTokenResponse;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class GraphApiUserManagerIdentityProviderMapper extends AbstractIdentityProviderMapper {
```

Comment:
Maybe nitpicking but I'd include `Microsoft` in the all of the names. `GraphApi....` isn't clear if it's Microsofts or e.g. Metas Graph API.